### ISSUE TITLE
Build instructions & provide builds

### ISSUE THREAD

**Issue by ORelio** [KoalaBR/spotlight#1]
Hi, thanks for this project! I'm looking for a way to automatically download Spotlight images for use on non-Windows 10 systems and it seems like that's the exact purpose of your project :+1:

Unfortunately, there are no build in the [releases](https://github.com/KoalaBR/spotlight/releases) section so potential users cannot download a zipfile with the application ready to use.

It would be nice to provide:

 - Build instructions for Windows and Linux
 - Windows builds in zip format
 - Maybe also `.deb` packages?

As I was curious enough to try it out, I managed to build it using the technical informations you provided in [Readme.md](https://github.com/KoalaBR/spotlight/blob/gh_remote/readme.md). Below are the steps I followed, which took several hours in order to install everything.

_______

_Tested on Windows 7 professional SP1, 64 bits_

1. Download Visual Studio Community 2017 on https://www.visualstudio.com/downloads/
2. Check both universal app and Desktop C++ development, click Install
3. Wait a LONG time for VS to download and install
4. Exit installer without launching Visual Studio

5. Open Spotlight.pro with Notepad++ or WordPad or VS text editor (NOT Notepad)
6. Locate the `INCLUDEPATH` line and update the Windows Kit version number
    E.g. in my case I went to `C:/Program Files (x86)/Windows Kits/10/Include/`
    and the folder inside was named `10.0.16299.0` so the line should be edited to read:
    `INCLUDEPATH += "C:/Program Files (x86)/Windows Kits/10/Include/10.0.16299.0/ucrt"`
7. Same step with the `LIBS` line just after `INCLUDEPATH`
8. Save and exit text editor

9. Download https://www.qt.io/download-qt-installer
10. While installing, check `Qt 5.9.2`
11. Wait a LONG time for Qt to download and install
12. Launch QT creator, open the `Spotlight.pro` file
13. Click "Manage" besides `Desktop Qt 5.9.2 MSVC2017 64bit`
14. Select `Desktop Qt 5.9.2 MSVC2017 64bit` in the compiler list
15. Define `Microsoft Visual C++ Compiler 15.0 (amd64)` for both C and C++, click OK
16. Select `Desktop Qt 5.9.2 MSVC2017 64bit`, click `Details`
17. Check `Release` only, define destination folder, then click `Configure Project`
18. From the `Build` menu, select `Build the "Spotlight" project`
19. Wait for compilation (progress bar is on bottom right of Qt creator)
20. Grab `Spotlight.exe` in destination folder defined in step 17, in subfolder "release"
21. Add the following DDLs in the same directory for Spotlight.exe to work properly:
    `C:\Qt\5.9.2\msvc2017_64\bin\Qt5Core.dll`
    `C:\Qt\5.9.2\msvc2017_64\bin\Qt5Gui.dll`
    `C:\Qt\5.9.2\msvc2017_64\bin\Qt5Sql.dll`
    `C:\Qt\5.9.2\msvc2017_64\bin\Qt5Widgets.dll`
    `C:\Qt\5.9.2\msvc2017_64\bin\Qt5Network.dll`

Note: Using MinGW without installing Visual Studio gave me the following error:

````
C:\Users\User\Desktop\spotlight-gh_remote\windowsdesktopsupport.cpp:25: error : invalid use of incomplete type 'IDesktopWallpaper {aka struct IDesktopWallpaper}'
         pDesktopWallpaper->SetWallpaper(nullptr, fname.toStdWString().c_str());
                          ^
````

**Comment by KoalaBR** [comment]
Ok, just some feedback: 

This is my first project published via github and / or for a wider audience than me or my friends. So I am learning on the fly and releases weren't high priority.

One of the next steps will be some changes to the .pro file in order to make building more flexible than the current version.

If this works out, releases would probably be the next step.

I am mainly using Linux (Qt-Creator) as my main development system so I would hope that doing releases there wouldn't be that hard.

For Windows (both 7 and 10) I am using Qt Creator too. However, I am using the Visual Studio 2017 64 Bit compilers. I may add project files for Visual Studio too but note that I would be using the Qt Tools for VS

I haven't used MinGW for this project but if you supply 32 Bit libraries for Openssl (see my answer to your issue #3 ) I wouldn't expect any problems.

Regarding .deb packages: As I am using Devuan (a systemd-less Debian fork) I would be interested. But I have absolutely no idea how to do this, any pointers?

Thanks alot
Koala

**Comment by ORelio** [comment]
Hi,

No worries, everyone has to start with something, that's also why I tried to give as much detail as possible in the issues I opened. Take your time 👍

Regarding .deb packages, actually I never had to build a package myself but I assume Debian packages would also work on Devuan and Ubuntu. So maybe the official documentation from Debian would help: https://wiki.debian.org/HowToPackageForDebian ... anyway, I guess that having simple build instructions for Linux would be a prerequiste before making a package.
